http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/internal/RepeatableInputStreamRequestEntity.java

```diff
@@ -109,7 +110,8 @@ private long parseContentLength(String contentLength) {
      * @return The request content input stream or an empty input stream if there is no content.
      */
     private InputStream getContent(SdkHttpFullRequest request) {
-        return request.content().orElseGet(() -> new ByteArrayInputStream(new byte[0]));
+        return request.contentStreamProvider().map(ContentStreamProvider::reset)
```

Comment:
Can we keep the orElseGet so we don't create the stream if we don't need it.

Reply:
+1 done.